modules/extension/xsd/xsd-wmts/src/main/java/org/geotools/wmts/bindings/TileMatrixSetBinding.java

```diff
@@ -120,7 +120,11 @@ public Object parse(ElementInstance instance, Node node, Object value) throws Ex
         matrixSet.setBoundingBox((BoundingBoxType) node.getChildValue("BoundingBox"));
         matrixSet.setIdentifier((CodeType) node.getChildValue("Identifier"));
         matrixSet.setSupportedCRS(((URI) node.getChildValue("SupportedCRS")).toString());
-        matrixSet.setWellKnownScaleSet((String) node.getChildValue("WellKnownScaleSet"));
+
+        URI wkss = (URI)node.getChildValue("WellKnownScaleSet");
+        if(wkss != null) {
```

Comment:
Formatting, the if should be "if (wkss != null)" (with a space). I make this mistake a lot myself, @dromagnoli catches it every time :-p